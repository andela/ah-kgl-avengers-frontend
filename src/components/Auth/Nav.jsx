import React from 'react';
import { Link } from 'react-router-dom';
import Logo from '../../assets/ah_logo_A1_Rectangle_54_pattern.png';

export default function Title() {
  return (
    <nav className="nav">
      <Link to="/" class="nav-logo left">
        <img
          src={Logo}
          alt=""
          srcSet=""
        />
        <img
          src=""
          alt=""
          srcSet=""
        />
      </Link>
    </nav>
  );
}
