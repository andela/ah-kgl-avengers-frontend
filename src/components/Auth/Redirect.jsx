import React, { Component } from 'react';
import {
  Card,
} from 'reactstrap';

class Redirect extends Component {
  state = {
  }

  render() {
    return (
      <div className="auth-container">
        <Card body className="col-sm-8 col-md-6 col-lg-4 mt-5 w3-card shadow-lg">
          <h3>Email Confirmation</h3>
          <p>
          You can Reach to the inbox of your email you have provided,
          We have sent you the confirmation email
          </p>
        </Card>
      </div>
    );
  }
}

export default Redirect;
