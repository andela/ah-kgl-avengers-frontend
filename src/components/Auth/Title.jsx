import React from 'react';
import {
  Button, Form, FormGroup, Label, Input,
} from 'reactstrap';
import { Link } from 'react-router-dom';

export default function Title() {
  return (
    <Form>
      <h3 className="card-title">
        sign
        <span>up</span>
      </h3>
      <FormGroup>
        <Label for="exampleEmail">Username</Label>
        <Input type="username" name="username" placeholder="snow" />
      </FormGroup>
      <FormGroup>
        <Label for="exampleEmail">Email</Label>
        <Input type="email" name="email" placeholder="snowice@company.com" />
      </FormGroup>
      <FormGroup>
        <Label for="exampleEmail">Password</Label>
        <Input type="password" name="password" placeholder="password" pattern="[a-zA-Z0-9]" />
      </FormGroup>
      <FormGroup>
        <Label for="exampleEmail">Confirm Password</Label>
        <Input type="password" name="password" placeholder="password" pattern="[a-zA-Z0-9]" />
      </FormGroup>
      <div className="card-action">
        <Button>Signup</Button>
      </div>
      <div className="card-spacer" />
      <div className="card-footer">
        <p>
          Already have an account?
          <a href="../signinTemplate/signin.html">sign in</a>
        </p>
        <p>
          By signing up you agree to all
          <Link to="/"> Terms and conditions</Link>
        </p>
      </div>
    </Form>
  );
}
