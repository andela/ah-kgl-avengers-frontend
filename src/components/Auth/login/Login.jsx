import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import {
  Form, FormGroup, Label, Input,
} from 'reactstrap';
import { Link } from 'react-router-dom';
import Side from '../Side';
import img from '../../../assets/developers_2_A1_Rectangle_58_pattern.png';

class Login extends Component {
  state = {
    username: '',
    password: ''
  }

  onChange = (e) => {
    const { onInputChange } = this.props;
    onInputChange({ field: e.target.name, value: e.target.value });
  }

  render() {
    return (
      <Fragment>
        <div className="wrapper">
          <section className="col-12 col-lg-8 col-md-10 col-sm-10">
            <div className="container">
              <div className="row">
                <div className="col-12 col-md-10 col-lg-7 signup-container text-center">
                  <Side img={img} />
                </div>
                <div className="col-12 col-md-10 col-lg-5 sign-up">
                  <Form onSubmit={this.onSubmit}>
                    <h3 className="sign-up-title mt-4 text-center">
                      sign
                      <span>in</span>
                    </h3>
                    <FormGroup>
                      <Label className="label-color">Email</Label>
                      <Input
                        className="input-color"
                        type="email"
                        name="email"
                        placeholder="snowice@company.com"
                        onChange={this.onChange}
                        value="email"
                      />
                    </FormGroup>
                    <FormGroup>
                      <Label className="label-color">Password</Label>
                      <Input
                        className="input-color"
                        type="password"
                        name="password1"
                        placeholder="password"
                        pattern="[a-zA-Z0-9]"
                        onChange={this.onChange}
                        value="password"
                      />
                    </FormGroup>
                    <div className="text-center mb-4">
                      <button type="button" className="sign-up-btn">Login</button>
                    </div>

                    <div className="sign-up-footer">
                      <p>
                        Don't have an account?
                        <a href="/sign-up"> sign up</a>
                      </p>
                      <p>
                        By signing up you agree to all
                        <Link to="/"> Terms and conditions</Link>
                      </p>
                    </div>
                  </Form>
                </div>
              </div>
            </div>
          </section>

        </div>
      </Fragment>
    );
  }
}

export const mapDispatchToProps = (dispatch, ownProps) => ({

});

export const mapStateToProps = state => ({

});

export default connect(mapStateToProps, mapDispatchToProps)(Login);
