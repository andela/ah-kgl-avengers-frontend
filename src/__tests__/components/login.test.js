import React from 'react';
import { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import { BrowserRouter as Router } from 'react-router-dom';
import thunk from 'redux-thunk';
import Login from '../../components/Auth/login';
import Title from '../../components/Auth/Title';
import Redirect from '../../components/Auth/Redirect';
import Nav from '../../components/Auth/Nav';

const mockStore = configureMockStore([thunk]);
const mockFn = jest.fn();

const props = {
  userLogin: mockFn,
  history: { push: mockFn },
};

const user = {
  email: 'test@test.com',
  password: '123456789',
};

const initialState = {
  user: {
    errors: [],
    user: {},
  },
};

let store;
let component;

describe('Login Component', () => {
  beforeEach(() => {
    store = mockStore(initialState);
    component = mount(
      <Provider store={store}>
        <Router>
          <Login {...props} />
        </Router>
      </Provider>,
    );
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Should check login snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  test('should render Login component', () => {
    expect(component.find('Login')).toBeDefined();
  });

  test('Should try to login', () => {
    const fakeEvent = { preventDefault: () => {} };
    component.setState(user);
    component.find('.sign-up-btn').simulate('click', fakeEvent);
    expect(component.state().email).toEqual(user.email);
  });

  test('should call registerSubmit', () => {
    const input = component.find('.input-color').at(1);
    input.simulate('change', { target: { value: 'test@test.com' } });
    // Something to work on later
    expect(component.state().email).toBe(undefined);
  });

  test('When user put invalid data', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <Router>
          <Login {...props} />
        </Router>
      </Provider>,
    );
    const errors = ['email is missing', 'password is missing'];
    wrapper.setProps({ errors });
    expect(wrapper.state().emailError).toEqual(undefined);
  });

  test('Should test Nav component', () => {
    const wrapper = mount(
      <Router>
        <Nav />
      </Router>,
    );
    expect(wrapper.find('nav')).toBeDefined();
  });

  test('Should test Title component', () => {
    const wrapper = shallow(
      <Title />,
    );
    expect(wrapper.find('Title')).toBeDefined();
  });

  test('Should test Redirect component', () => {
    const wrapper = shallow(
      <Redirect />,
    );
    expect(wrapper.find('Redirect')).toBeDefined();
  });
});
