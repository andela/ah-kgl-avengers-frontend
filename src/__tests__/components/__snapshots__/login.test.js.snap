// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Register /> should render without crashing 1`] = `ShallowWrapper {}`;
